### ISSUE TITLE
Type stability using `convert`

### ISSUE THREAD

**Issue by nandoconde** [JuliaMath/Polynomials.jl#372]
The conversion from `ChebyshevT` to `Polynomial` does not return the numeric type expected.
I don't know if it is desired behavior or a bug. See caveat at the end.

MWE:
```julia
# Create T2(x) coefficients
p_coeffs = zeros(BigFloat, n);
p_coeffs[n] = one(BigFloat);

# Create T2(x) using ChebyshevT
P_c = ChebyshevT(p_coeffs, :x)

# Convert to Polynomial
P_x = convert(Polynomial, P_c)

# Different element types
eltype(P_c)    # BigFloat
eltype(P_x)    # Float64
```

**Caveat**
Please, do note that `P_x = convert(Polynomial{BigFloat}, P_c)` does return the desired type.

**Comment by jverzani** [comment]
Thanks! This shouldn't be too hard to fix.

**Comment by jverzani** [comment]
Thanks again. This was addressed in #373.

**Comment by nandoconde** [comment]
Thank you very much!! That was fast, you are awesome!
